File: app/src/androidTest/java/com/example/task2/TestBenchmarkModule.java
package com.example.task2;

import com.example.task2.models.benchmarks.Benchmark;
import com.example.task2.models.benchmarks.FakeCollectionBenchmark;
import com.example.task2.models.benchmarks.FakeMapBenchmark;

import javax.inject.Named;

import dagger.Module;
import dagger.Provides;

@Module
public class TestBenchmarkModule {

    @Provides
    @Named("CollectionBenchmark")
    public Benchmark provideCollectionBenchmark() {
        return new FakeCollectionBenchmark();
    }

    @Provides
    @Named("MapBenchmark")
    public Benchmark provideMapBenchmark() {
        return new FakeMapBenchmark();
    }
}

File: app/src/androidTest/java/com/example/task2/models/TestBenchmarkModule.kt
package com.example.task2.models

import com.example.task2.models.benchmarks.Benchmark
import com.example.task2.models.benchmarks.FakeCollectionBenchmark
import com.example.task2.models.benchmarks.FakeMapBenchmark
import dagger.Module
import dagger.Provides
import javax.inject.Named

@Module
class TestBenchmarkModule {

    @Provides
    @Named("CollectionBenchmark")
    fun provideCollectionBenchmark(): Benchmark = FakeCollectionBenchmark()

    @Provides
    @Named("MapBenchmark")
    fun provideMapBenchmark(): Benchmark = FakeMapBenchmark()
}
